archive/tar_path.go

```diff
@@ -0,0 +1,26 @@
+package archive
+
+import (
+	"path/filepath"
+	"runtime"
+	"strings"
+)
+
+// TarPath converts an OS path to a path suitable for TAR headers
+func TarPath(path string) string {
+	volumeName := filepath.VolumeName(path)
+	path = strings.TrimPrefix(path, volumeName)
+	return filepath.ToSlash(path)
```

Comment:
Interesting question... Originally, this wasn't based on any concrete evidence other than the patterns in our code for both `pack` and `lifecycle`. We chose to "normalize" the creation of tars to simplify the code (and because somewhere along the way it was stuck in our heads that tar headers always used forward slashes).

Looking at the TAR spec seems to be consistent...

For the slashes (vs backslashes), I found some evidence for that [here](https://www.gnu.org/software/tar/manual/html_node/Standard.html):
> The `name` field is the file name of the file, with directory names (if any) preceding the file name, separated by slashes.

The volume part is harder to nail down in the TAR spec as it's not mentioned in the above quote. Based on pages like [this](https://www.gnu.org/software/tar/manual/html_node/absolute.html), it doesn't seem like volume names would be expected in the `name` field though. I'd imagine having a volume name in the header path would make untarring just have to do the same trimming.